### ISSUE TITLE
Player is not actually removed when they quit.

### ISSUE THREAD

**Issue by NgLamVN** [Ifera/ScoreHud#176]
# Crashlog:
```
17:04:49 Server|CRITICAL - BadFunctionCallException: "Cannot set a score to a player without a scoreboard. Please call ScoreFactory::setScore() beforehand." (EXCEPTION) in "plugins/ScoreHud.phar/src/Ifera/ScoreHud/libs/jackmd/scorefactory/ScoreFactory" at line 83
17:04:49 Server|CRITICAL - #0 plugins/ScoreHud.phar/src/Ifera/ScoreHud/scoreboard/Scoreboard(180): Ifera\ScoreHud\libs\jackmd\scorefactory\ScoreFactory::setScoreLine(object JackMD\NameSpaceRemover\CustomPlayer, integer 1, string[29]  ..2.....l..eWelcome:..r ..f )
17:04:49 Server|CRITICAL - #1 plugins/ScoreHud.phar/src/Ifera/ScoreHud/EventListener(110): Ifera\ScoreHud\scoreboard\Scoreboard->display()
17:04:49 Server|CRITICAL - #2 plugins/ScoreHud.phar/src/Ifera/ScoreHud/EventListener(92): Ifera\ScoreHud\EventListener->updateTag(object JackMD\NameSpaceRemover\CustomPlayer, object Ifera\ScoreHud\scoreboard\ScoreTag)
17:04:49 Server|CRITICAL - #3 plugins/ScoreHud.phar/src/Ifera/ScoreHud/EventListener(71): Ifera\ScoreHud\EventListener->updateServerTag(object Ifera\ScoreHud\scoreboard\ScoreTag)
17:04:49 Server|CRITICAL - #4 pmsrc/src/pocketmine/plugin/MethodEventExecutor(42): Ifera\ScoreHud\EventListener->onServerTagUpdate(object Ifera\ScoreHud\event\ServerTagUpdateEvent)
17:04:49 Server|CRITICAL - #5 pmsrc/src/pocketmine/plugin/RegisteredListener(80): pocketmine\plugin\MethodEventExecutor->execute(object Ifera\ScoreHud\EventListener, object Ifera\ScoreHud\event\ServerTagUpdateEvent)
17:04:49 Server|CRITICAL - #6 pmsrc/src/pocketmine/event/Event(88): pocketmine\plugin\RegisteredListener->callEvent(object Ifera\ScoreHud\event\ServerTagUpdateEvent)
17:04:49 Server|CRITICAL - #7 plugins/BasicScore_dev-8.phar/src/Ifera/BasicScore/listeners/EventListener(38): pocketmine\event\Event->call()
17:04:49 Server|CRITICAL - #8 pmsrc/src/pocketmine/scheduler/ClosureTask(61): Ifera\BasicScore\listeners\EventListener->Ifera\BasicScore\listeners\{closure}(integer 414)
17:04:49 Server|CRITICAL - #9 pmsrc/src/pocketmine/scheduler/TaskHandler(137): pocketmine\scheduler\ClosureTask->onRun(integer 414)
17:04:49 Server|CRITICAL - #10 pmsrc/src/pocketmine/scheduler/TaskScheduler(179): pocketmine\scheduler\TaskHandler->run(integer 414)
17:04:49 Server|CRITICAL - #11 pmsrc/src/pocketmine/plugin/PluginManager(654): pocketmine\scheduler\TaskScheduler->mainThreadHeartbeat(integer 414)
17:04:49 Server|CRITICAL - #12 pmsrc/src/pocketmine/Server(2487): pocketmine\plugin\PluginManager->tickSchedulers(integer 414)
17:04:49 Server|CRITICAL - #13 pmsrc/src/pocketmine/Server(2262): pocketmine\Server->tick()
17:04:49 Server|CRITICAL - #14 pmsrc/src/pocketmine/Server(2105): pocketmine\Server->tickProcessor()
17:04:49 Server|CRITICAL - #15 pmsrc/src/pocketmine/Server(1689): pocketmine\Server->start()
17:04:49 Server|CRITICAL - #16 pmsrc/src/pocketmine/PocketMine(291): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[37] D:\MinecraftTool\FarmingIslandServer\, string[45] D:\MinecraftTool\FarmingIslandServer\plugins\)
17:04:49 Server|CRITICAL - #17 pmsrc/src/pocketmine/PocketMine(321): pocketmine\server()
17:04:49 Server|CRITICAL - #18 pmsrc(11): require(string[92] phar://D:/MinecraftTool/FarmingIslandServer/PocketMine-MP.phar/src/pocketmine/Po)
17:04:49 Server|EMERGENCY - An unrecoverable error has occurred and the server has crashed. Creating a crash dump
17:04:49 Server|EMERGENCY - Please upload the "D:\MinecraftTool\FarmingIslandServer\crashdumps/Mon_May_3-17.04.49-WIB_2021.log" file to the Crash Archive and submit the link to the Bug Reporting page. Give as much info as you can.
```
# Crashdump
[Mon_May_3-17.04.49-WIB_2021.log](https://github.com/Ifera/ScoreHud/files/6414243/Mon_May_3-17.04.49-WIB_2021.log)
# default-board format:
```
- "Online: {basicscore.online}"
```

**Comment by Ifera** [comment]
Can you please give me a breakdown on when this issue occurs?

**Comment by NgLamVN** [comment]
> Can you please give me a breakdown on when this issue occurs?

When you exit server, the server will crash after 1 seconds.

**Comment by Ifera** [comment]
> When you exit server, the server will crash after 1 seconds.

Oh alright, i think i know what's causing it, will have it fixed in a bit.

**Comment by Ifera** [comment]
Please try using dev build 63 [here](https://poggit.pmmp.io/r/124750/ScoreHud_dev-63.phar) and let me know if this fixes the issue.

**Comment by NgLamVN** [comment]
> Please try using dev build 63 [here](https://poggit.pmmp.io/r/124750/ScoreHud_dev-63.phar) and let me know if this fixes the issue.

Yep, it work. This issue is fixed !

**Comment by Ifera** [comment]
Aright that's great. Please if you found another issue then don't hesitate to make a new issue. Thanks!

**Comment by Ifera** [comment]
Fixed in [b8e4546](https://github.com/Ifera/ScoreHud/commit/b8e4546709fd675456b8c60a6fb7d99aa3901680)
